ehcache-impl/src/main/java/org/ehcache/impl/internal/store/tiering/TieredStore.java

```diff
@@ -572,7 +572,10 @@ public NoopCachingTier(final AuthoritativeTier<K, V> authoritativeTier) {
     @Override
     public ValueHolder<V> getOrComputeIfAbsent(final K key, final Function<K, ValueHolder<V>> source) {
       final ValueHolder<V> apply = source.apply(key);
-      authoritativeTier.flush(key, apply);
+      //this null check is to handle the null pointer exception that can be thrown from the flush method
```

Comment:
Can we phrase this more concretely, and maybe without using a negation. Something like:
```java
if (apply != null) {
  //immediately flush any entries faulted from the authority as this tier has no capacity
  authoritativeTier.flush(key, apply);
}
```